src/application/Application.mjs

```diff
@@ -102,12 +102,31 @@ export default class Application extends Component {
         this.__updateFocus();
     }
 
+    __focusedComponent() {
+        return this._focusPath ? this._focusPath[this._focusPath.length - 1] : undefined;
+    }
+
     __updateFocus() {
+        const prevFocusedComponent = this.__focusedComponent();
+
+        // The Application's focus path, therefore final focused component, may change here...
         const notOverridden = this.__updateFocusRec();
 
         if (!Application.booting && notOverridden) {
             this.updateFocusSettings();
         }
+
+        if (this.__options.debug) {
+            const newFocusedComponent = this.__focusedComponent();
+
+            // Testing for updateDebugFocusState safeties against lightning inspector not having been attached.
+            prevFocusedComponent && prevFocusedComponent.updateDebugFocusState && prevFocusedComponent.updateDebugFocusState();
+            newFocusedComponent && newFocusedComponent.updateDebugFocusState && newFocusedComponent.updateDebugFocusState();
+
+            if (prevFocusedComponent !== newFocusedComponent) {
+                console.log('FOCUS ' + (newFocusedComponent && newFocusedComponent.getLocationString()));
```

Comment:
Please add `[Lightning]` tag to the log. This distinction is helpful when there are other logging mechanisms used on top.

Reply:
@jhderojasUVa if this console log doesn't shadow one that was in the original code, maybe we can just remove this if clause?